src/session.rs

```diff
@@ -1057,3 +1045,81 @@ impl SessionState {
         }
     }
 }
+
+struct InnerSocket<T> {
+    socket: YamuxSession<T>,
+    sender: priority_mpsc::Sender<SessionEvent>,
+}
+
+impl<T> InnerSocket<T>
+where
+    T: AsyncRead + AsyncWrite + Unpin + Send,
+{
+    fn new(socket: YamuxSession<T>, sender: priority_mpsc::Sender<SessionEvent>) -> Self {
+        InnerSocket { socket, sender }
+    }
+}
+
+impl<T> Stream for InnerSocket<T>
+where
+    T: AsyncRead + AsyncWrite + Unpin,
+{
+    type Item = ();
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Option<Self::Item>> {
+        match Pin::new(&mut self.socket).as_mut().poll_next(cx) {
+            Poll::Ready(Some(Ok(stream))) => {
+                let mut sender = self.sender.clone();
+
+                tokio::spawn(async move {
+                    let _ignore = sender.send(SessionEvent::StreamStart { stream }).await;
```

Comment:
Better `log::debug` the error.

Reply:
I don’t think it needs to be dealt with. If something goes wrong here, there is only one possibility that the channel is closed. If it is closed, it is the right thing to silently close the corresponding stream

Reply:
It's mainly for debug purpose

Reply:
I don’t agree with this. What is the point of adding debug in a place where the return value is fully known?